From 62e6a1c2d38183193268dc75318dde24e43dd742 Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Mon, 9 Sep 2019 14:33:51 +0800
Subject: [PATCH] DRVLIB_SW-2846 - Fix klocwork issue in qspi driver

Initialize return value to handle path where spi xfer length is 0.
---
 drivers/spi/spi-cadence-qspi-apb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-cadence-qspi-apb.c b/drivers/spi/spi-cadence-qspi-apb.c
index 68cd215336bd..524b7961dc20 100644
--- a/drivers/spi/spi-cadence-qspi-apb.c
+++ b/drivers/spi/spi-cadence-qspi-apb.c
@@ -616,7 +616,7 @@ static int cadence_qspi_apb_indirect_write_execute(
 	struct struct_cqspi *cadence_qspi, unsigned txlen,
 	const unsigned char *txbuf, unsigned int flash_type)
 {
-	int ret;
+	int ret = 0;
 	unsigned int timeout;
 	unsigned int reg = 0;
 	unsigned int *irq_status = &(cadence_qspi->irq_status);
